src/FSharp.Core/seqcore.fs

```diff
@@ -387,6 +387,70 @@ module RuntimeHelpers =
     let EnumerateThenFinally (source: seq<'T>) (compensation: unit -> unit)  =
         (FinallyEnumerable(compensation, (fun () -> source)) :> seq<_>)
 
+
+    let EnumerateTryWith (source : seq<'T>) (exceptionFilter:exn -> int) (exceptionHandler:exn -> seq<'T>) =
+        let originalSource = lazy(source.GetEnumerator())
+        let mutable shouldDisposeOriginalAtTheEnd = true
+        let mutable exceptionalSource : option<IEnumerator<'T>> = None     
```

Comment:
nitpick: Should probably be either `Option<IEnumerator<'T>` or, more idiomatic `IEnumerator<'T> option`.

Reply:
Agree, will apply once github allows it (right now it says review is pending)